polyfill/lib/ecmascript.mjs

```diff
@@ -951,23 +956,38 @@ export const ES = ObjectAssign({}, ES2020, {
     if (validUnits.indexOf(unit1) > validUnits.indexOf(unit2)) return unit2;
     return unit1;
   },
-  ToPartialRecord: (bag, fields, cast = ES.ToInteger) => {
+  CastIfDefined: (value, cast) => {
+    if (value !== undefined) {
+      return cast(value);
+    }
+    return value;
+  },
+  ToPartialRecord: (bag, fields, callerCast) => {
     if (ES.Type(bag) !== 'Object') return false;
     let any;
     for (const property of fields) {
       const value = bag[property];
       if (value !== undefined) {
         any = any || {};
-        if (BUILTIN_FIELDS.has(property)) {
-          any[property] = cast(value);
+        if (callerCast === undefined && BUILTIN_CASTS.has(property)) {
+          any[property] = BUILTIN_CASTS.get(property)(value);
+        } else if (callerCast !== undefined) {
+          any[property] = callerCast(value);
         } else {
           any[property] = value;
         }
       }
     }
     return any ? any : false;
   },
-  ToRecord: (bag, fields) => {
+  ToPositiveInteger: (value, property) => {
+    value = ToInteger(value);
+    if (value < 1) {
+      throw new RangeError(`property '${property}' cannot be a negative integer or zero`);
```

Comment:
nit: "cannot be a negative integer" is slightly incongruous if you passed in `-3.5`

Reply:
```suggestion
      throw new RangeError(`property '${property}' must be a positive integer`);
```

Reply:
Can be NaN or +Infinity, so 'must be a positive integer' isn't quite right. How about:
'cannot be a number less than one'